### PR TITLE
Message appears when trying to execute/export program with syntax errors

### PR DESCRIPTION
**Pull request by djmcalleja**
Fixes https://github.com/softdevteam/eco/issues/151

### CODE REVIEW

**Comment by snim2** [comment]
Great! BTW if you say "Fixes Issue ..." then GitHub will automatically close the issue when this has been merged.

File: lib/eco/debug.py [hunk 1]
```diff
@@ -13,22 +13,23 @@ def __init__(self, window):
...
         self.line_read_until = '(Pdb)'   
         self.process = self.window.getEditor().tm.export(debug=True)
-        flags = fcntl(self.process.stdout, F_GETFL)
-        fcntl(self.process.stdout, F_SETFL, flags | O_NONBLOCK)       
         if not self.process:
-            self.emit(self.signal_output, "Cannot run debugger process, cannot debug.")
+            self.emit(self.signal_execute_fail)
             self.exit()
             return
+        flags = fcntl(self.process.stdout, F_GETFL)
```
Comment (vext01) [inline lib/eco/debug.py:31]: I've never spent much time on eco, so here's some dumb questions:

Why does popping up an error require changes to file descriptor flags?

File: lib/eco/eco.py [hunk 2]
```diff
@@ -891,7 +892,9 @@ def cli_export(self, source, dest, fast):
             self.tm.fast_export(language_boxes, dest, source=source)
         else:
             self.tm.load_file(language_boxes)
-            self.tm.export(dest, source=source)
+            ex=self.tm.export(dest, source=source)
```
Comment (vext01) [inline lib/eco/eco.py]: you need spaces around that assignment.

File: lib/eco/eco.py [hunk 3]
```diff
@@ -1174,7 +1179,12 @@ def exportAs(self):
         if path:
             self.save_last_dir(str(path))
             ed.export_path = path
-            self.getEditor().tm.export(path, source=ed.filename)
+            ex=self.getEditor().tm.export(path, source=ed.filename)
```
Comment (vext01) [inline lib/eco/eco.py]: and here

**Comment by vext01** [comment]
@ptersilie a lot of this is QT event loop stuff, does it look OK to you?

**Comment by ltratt** [comment]
@vext01 For these sorts of PRs, I think you have to run them locally to check.

**Comment by vext01** [comment]
I mean in terms of code quality. I am going to try it out though.

File: lib/eco/debug.py (hunk 1 above)
Comment (djmcalleja) [inline lib/eco/debug.py:31]: The line hasn't changed but I had to move it to after line 28, which checks if `self.process` has a value

Comment (ltratt) [inline lib/eco/debug.py:31]: Note that this is a standard fcntl idiom -- you need to read the flags before setting it (in the next line) to non-blocking (maintaining other flags).

**Comment by vext01** [comment]
FWIW, this works from a functional standpoint.

File: lib/eco/debug.py (hunk 1 above)
Comment (vext01) [inline lib/eco/debug.py:31]: Yes, I know fcntl ;)

I think this is OK.

**Comment by vext01** [comment]
Looks OK from where I am standing, but I think you should ask Lukas, as I don't really know what I'm doing when it comes to QT.

**Comment by ltratt** [comment]
If it runs OK, and the patch doesn't look insane, I think we're OK :)

**Comment by vext01** [comment]
Alright then :P

**Comment by ltratt** [comment]
Um, this stuff still needed squashing. Too late now though...

**Comment by vext01** [comment]
Yeah, I accidentally hit merge. Sorry.

**Comment by ptersilie** [comment]
@vext01 I think for most PRs you can squash during merging on Github itself.

**Pull request merged**